server/src/main/java/org/elasticsearch/index/analysis/IndexAnalyzers.java

```diff
@@ -160,6 +167,23 @@ public List<String> reload(AnalysisRegistry registry, IndexSettings indexSetting
 
                 return reloadableAnalyzers.stream().map(NamedAnalyzer::name).toList();
             }
+
+            /**
+             * Initialize analyzers from external resources if applicable.
+             * This is called during index creation in shard recovery.
+             */
+            @Override
+            public void initializeFromResources() {
+                // we need to do initialization only once, even if this is called by multiple shards,
+                // as all shards of the same index share the same analyzers
+                if (initialized) return;
```

Comment:
Would it make sense to use a semaphore as well to avoid multiple threads doing the initialization before at least one is finished?